src/gc/gcpriv.h

```diff
@@ -3002,6 +3002,17 @@ class gc_heap
     PER_HEAP_ISOLATED
     size_t min_segment_size;
 
+    // For SOH we always allocate segments of the same
+    // size unless no gc region requires larger ones.
```

Comment:
Yes, that works too. The way it is now, it sounds like something totally different.